src/Suave/Suave.fsproj

```diff
@@ -140,15 +179,7 @@
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Xml" />
   </ItemGroup>
-  <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">
-      <ItemGroup>
-        <Reference Include="FSharp.Core">
-          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
+  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">
+    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2 project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

```

Comment:
Please don’t be hard on the nuget and framework tools team, this stuff has changed constantly and broke me at least 3 times of the last 4 months.  One thing that has become apparent is that the world is used to us shipping great tools in 2 or three year chunks, the constant iterating we have always done internally is difficult to manage with external partners.  I am used to it, having been here for ages and so don’t even find it frustrating.  The way I deal with it is to trust the outcome is going to be great and then focus on the minimal set of information I need to move my work forward.

It is not always comfortable but the result is usually pretty good.

If you have specific questions, email me directly and I will find out exactly who to ask to get the gospel.  However, be aware that the Gospel may change on a frequent cadence ☺


Kevin

From: Steffen Forkmann [mailto:notifications@github.com]
Sent: Thursday, November 12, 2015 10:49 AM
To: SuaveIO/suave <suave@noreply.github.com>
Cc: Kevin Ransom <Kevin.Ransom@microsoft.com>
Subject: Re: [suave] Enable build and run on coreclr for suave (#334)


In src/Suave/Suave.fsproj<https://github.com/SuaveIO/suave/pull/334#discussion_r44697369>:

> @@ -140,15 +179,7 @@

>      <Reference Include="System.Web.Extensions" />

>      <Reference Include="System.Xml" />

>    </ItemGroup>

> -  <Choose>

> -    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">

> -      <ItemGroup>

> -        <Reference Include="FSharp.Core">

> -          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>

> -          <Private>True</Private>

> -          <Paket>True</Paket>

> -        </Reference>

> -      </ItemGroup>

> -    </When>

> -  </Choose>

> +  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">

> +    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2<https://www.myget.org/F/dotnet-core;https:/www.myget.org/F/fsharp-daily;https:/nuget.org/api/v2> project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

Kevin, the point is we're trying to figure out what's needed in paket. Currently we didn't get good support from your colleagues. This pull request is pure gold since it creates a working F# app and I'm trying to understand what was needed to make it work. It's absolutely amazing stuff.
My questions are no criticism to this pull request. It's the opposite.

—
Reply to this email directly or view it on GitHub<https://github.com/SuaveIO/suave/pull/334/files#r44697369>.